sympy/algebras/quaternion.py

```diff
@@ -720,3 +720,389 @@ def to_rotation_matrix(self, v=None):
+
+        Examples
+        ========
+
+        >>> from sympy.algebras.quaternion import Quaternion
+
+        >>> q = Quaternion(4, 8, 13, 12)
+        >>> q.scalar_part()
+        4
+
+        """
+
+        return self.a
+
+    def vector_part(self):
+        """
+        Returns the vector part of the quaternion.
+
+        Explanation
+        ===========
+
+        If q is a quaternion given by q = a + b*i + c*j + d*k where a, b, c and d
+        are real numbers then the vector part of q is b*i + c*j + d*k.
+
+        Returns
+        =======
+
+        Quaternion: representing vector part of the quaternion.
+
+        Examples
+        ========
+
+        >>> from sympy.algebras.quaternion import Quaternion
+        >>> q = Quaternion(1, 1, 1, 1)
+        >>> q.vector_part()
+        0 + 1*i + 1*j + 1*k
+
+        >>> q = Quaternion(4, 8, 13, 12)
+        >>> q.vector_part()
+        0 + 8*i + 13*j + 12*k
+
+        """
+
+        return Quaternion(0, self.b, self.c, self.d)
+
+    def axis(self):
+        """
+        Returns the axis of the quaternion.
+
+        Explanation
+        ===========
+
+        The axis of a quaternion is the normalized versor (vector part) of that quaternion.
+
+        Returns
+        =======
+
+        Quaternion: representing axis of the provided quaternion.
+
+        Examples
+        ========
+
+        >>> from sympy.algebras.quaternion import Quaternion
+        >>> q = Quaternion(1, 1, 1, 1)
+        >>> q.axis()
+        0 + sqrt(3)/3*i + sqrt(3)/3*j + sqrt(3)/3*k
+
+        >>> q = Quaternion(4, 8, 13, 12)
+        >>> q.axis()
+        0 + 8*sqrt(377)/377*i + sqrt(377)/29*j + 12*sqrt(377)/377*k
+
+        See Also
+        ========
+
+        vector_part
+
+        """
+
+        return Quaternion(0, self.b, self.c, self.d).normalize()
+
+    def is_pure(self):
+        """
+        Returns true if the quaternion is pure else returns false.
+
+        Explanation
+        ===========
+
+        A pure quaternion (also a vector quaternion) is a quaternion with scalar
+        part equal to 0.
+
+        Examples
+        ========
+
+        >>> from sympy.algebras.quaternion import Quaternion
+        >>> q = Quaternion(1, 1, 1, 1)
+        >>> q.is_pure()
+        False
+
+        >>> q = Quaternion(0, 8, 13, 12)
+        >>> q.is_pure()
+        True
+
+        See Also
+        ========
+        scalar_part
+
+        """
+
+        if isinstance(self.a, Symbol):
+            z = self.a.is_zero
+            if not z:
+                return False
+            elif z == None:
+                return
+            else:
+                x = fuzzy_and(self.b.is_zero, self.c.is_zero,self.d.is_zero)
+                if x:
+                    return False
+                if x == None:
+                    return
+                else:
+                    return True
+        return self.a == 0  and not (self.a == self.b == self.c == self.d)
```

Comment:
This could just be
```
return fuzzy_and(
    self.a.is_zero,
    fuzzy_not(fuzzy_and(v.is_zero for v in [self.b, self.c, self.d]))
)
```
If we wanted a zero quaternion to be considered pure then it could just be
```
return self.a.is_zero
```
I know this was done by analogy with `is_imaginary` but personally I'd prefer it if zero was considered imaginary.